source/math/FloatPoint2DVectorList.ooc

```diff
@@ -107,4 +108,30 @@ FloatPoint2DVectorList: class extends VectorList<FloatPoint2D> {
 			result = result >> this[i] toString() >> "\n"
 		result
 	}
+	resampleLinear: func (start, end, interval: Float) -> This {
+		// Assumes list is sorted by x values.
+		resultCount := ((end - start) / interval) ceil() as Int + 1
+		result := This new()
+		previousIndex := 0
+		for (i in 0 .. resultCount) {
+			point := FloatPoint2D new(start + i * interval, 0.0f)
+			if (point x <= this[0] x)
+				point y = this[0] y
+			else if (point x >= this[this count - 1] x)
+				point y = this[this count - 1] y
+			else {
+				leftPoint, rightPoint: FloatPoint2D
+				for (j in previousIndex .. this count)
+					if (this[j] x <= point x) {
+						leftPoint = this[j]
+						rightPoint = this[Int minimum(j + 1, this count - 1)]
+						previousIndex = j
+					} else break
+				weight := Float absolute(point x - leftPoint x) / Float absolute(rightPoint x - leftPoint x)
```

Comment:
You never check if `point x` is smaller than `rightPoint x`, so you need to `clamp`. Don't ignore me! I bite you in the ass otherwise!